### PR TITLE
ci: Run tests on pull requests

### PR DESCRIPTION
**Pull request by morgsmccauley**
This PR adds a new Github Action to run tests on submitted pull requests.

### CODE REVIEW

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Pull Request
+
+on:
+  pull_request:
```
Comment (esaminu) [inline .github/workflows/pull-request.yml:4]: Would this also trigger the workflow for a push after opening the PR?

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  workflow_dispatch:
+
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (esaminu) [inline .github/workflows/pull-request.yml:10]: Would this only cancel previous runs of the same HEAD commit or all previous runs on the PR? I think we want the latter behavior. I think `github.ref` may work for that and there's also `github.event.pull_request.number`

**Review by esaminu — COMMENTED** [review]
LGTM but added some questions to confirm workflow trigger and cancel behavior 😃

File: .github/workflows/pull-request.yml (hunk 1 above)
Comment (esaminu) [reply .github/workflows/pull-request.yml:4]: Thanks for clarifying! 😃

Comment (andy-haynes) [reply .github/workflows/pull-request.yml:4]: Of course! I knew it worked from experience but I didn't think much about it until I saw your question so I went digging 😅

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (morgsmccauley) [reply .github/workflows/pull-request.yml:10]: The `group` here is what determines which workflows should be cancelled once a new one is triggered. Using `github.ref` (the branch name) will cancel all previous runs for the given PR, and combining with `github.workflow` (the name of this workflow) will ensure it only cancels _this_ workflow and not others triggered via the same PR.

**Comment by morgsmccauley** [comment]
Also I have intentionally left out eslint and TypeScript checks because there are lots of errors 😅 We can add those in once we get them to a stable state.

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (esaminu) [reply .github/workflows/pull-request.yml:10]: Nice 👍 was just not sure that different commits on the same branch have the same `github.ref`

**Comment by esaminu** [comment]
> Also I have intentionally left out eslint and TypeScript checks because there are lots of errors 😅 We can add those in once we get them to a stable state.

Cool, opinions of fuzz tests in this workflow?

**Comment by morgsmccauley** [comment]
> Cool, opinions on fuzz tests in this workflow?

@esaminu Do we have fuzz tests already in place for `near-api-js`?

**Review by esaminu — APPROVED** [review]
LGTM 💯

**Pull request merged**